model/sparql/GenericSparql.php

```diff
@@ -540,36 +540,40 @@ public function queryConceptInfo($uris, $arrayClass = null, $vocabs = array(), $
     /**
      * Generates the sparql query for queryTypes
      * @param string $lang
+     * @param string $vocid
      * @return string sparql query
      */
-    private function generateQueryTypesQuery($lang) {
-        $fcl = $this->generateFromClause();
+    private function generateQueryTypesQuery($lang, string $vocid = null): string {
+        $vocabs = isset($vocid) ? [$vocid] : $this->model->getVocabularies();
+        $fcl = $this->generateFromClause($vocabs);
         $query = <<<EOQ
 SELECT DISTINCT ?type ?label ?superclass $fcl
 WHERE {
-  {
-    { BIND( skos:Concept as ?type ) }
-    UNION
-    { BIND( skos:Collection as ?type ) }
-    UNION
-    { BIND( isothes:ConceptGroup as ?type ) }
-    UNION
-    { BIND( isothes:ThesaurusArray as ?type ) }
-    UNION
-    { ?type rdfs:subClassOf/rdfs:subClassOf* skos:Concept . }
-    UNION
-    { ?type rdfs:subClassOf/rdfs:subClassOf* skos:Collection . }
-  }
-  OPTIONAL {
-    ?type rdfs:label ?label .
-    FILTER(langMatches(lang(?label), '$lang'))
-  }
-  OPTIONAL {
-    ?type rdfs:subClassOf ?superclass .
-  }
-  FILTER EXISTS {
-    ?s a ?type .
-    ?s skos:prefLabel ?prefLabel .
+  GRAPH ?g {
```

Comment:
This looks wrong to me. You are already using `$fcl` i.e. a generated block of FROM clauses. There is no need for the `GRAPH ?g {` block.